src/Verifier/SAW/Simulator/What4.hs

```diff
@@ -596,12 +596,17 @@ parseUninterpreted sym ref nm args ty =
 mkUninterpreted ::
   forall sym args t. (IsSymExprBuilder sym) =>
   sym -> IORef (SymFnCache sym) ->
-  String -> Assignment (SymExpr sym) args -> BaseTypeRepr t ->
+  String {- ^ function name -} ->
+  Assignment (SymExpr sym) args ->
+  Assignment BaseTypeRepr args ->
+  BaseTypeRepr t ->
   IO (SymExpr sym t)
-mkUninterpreted sym ref nm args ret =
-  do fn <- mkSymFn sym ref nm (fmapFC W.exprType args) ret
+mkUninterpreted sym ref nm args tys ret =
+  do fn <- mkSymFn sym ref nm tys ret
      W.applySymFn sym fn args
 
+data SomeArgs f =
```

Comment:
nit: maybe `SomeSValue`, `nm` is not part if the arguments

Reply:
Yeah, I named the `SomeArgs` datatype before I decided to move the string into it. It's probably worthwhile to reconsider the names of a few internal functions and make sure the comments are still accurate.